signcrypt_seal.go

```diff
@@ -195,7 +196,15 @@ func (r ReceiverSymmetricKey) makeReceiverKeys(ephemeralPriv BoxSecretKey, paylo
 // receivers. Check that receivers aren't sent to twice; check that
 // there's at least one receiver and not too many receivers.
 func checkSigncryptReceivers(receiverBoxKeys []BoxPublicKey, receiverSymmetricKeys []ReceiverSymmetricKey) error {
-	receiverCount := len(receiverBoxKeys) + len(receiverSymmetricKeys)
+	// Handle possible (but unlikely) overflow when adding
+	// together the two sizes.
+	if uint32(len(receiverBoxKeys)) > maxReceiverCount {
+		return ErrBadReceivers
+	}
+	if uint32(len(receiverSymmetricKeys)) > maxReceiverCount {
+		return ErrBadReceivers
+	}
```

Comment:
Apart from checking the two variables independently, do we also need to check that the sum doesn't overflow?

Reply:
We already do! Right below. Ugh, but I just noticed that this check is buggy, since `int` may be 64 bits. Will fix...